workspaces/optic-engine/src/shapes/traverser.rs

```diff
@@ -129,23 +129,23 @@ impl<'a> Traverser<'a> {
           .map(|choice| {
             if let ShapeKind::ObjectKind = &choice.core_shape_kind {
               // - find field node by key in object's field node edges
-              let field_ids = self
+              let field_nodes = self
                 .shape_queries
-                .resolve_shape_field_id_and_names(&choice.shape_id);
+                .resolve_shape_fields_nodes(&choice.shape_id, false);
```

Comment:
In this case, traversing the shape for generating a diff should _not_ query removed shapes